tests/unit/common/middleware/test_container_hierarchy.py

```diff
@@ -344,7 +345,8 @@ def _test_listing_with_marker(self):
         resp = self.call_ch(req)
         names = [item.get('name', item.get('subdir'))
                  for item in json.loads(resp[2])]
-        self.assertNotIn('d1/', names)
+        self.assertNotIn('d0/', names)
+        self.assertIn('d1/', names)
```

Comment:
If the marker is `d1/`, it is not supposed to appear in the results.
[OpenStack documentation](https://developer.openstack.org/api-ref/object-store/index.html?expanded=show-container-details-and-list-objects-detail#containers) says:
> For a string value, x , constrains the list to items whose names are greater than x.